src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/StateSynchronizationPerformanceParameters.cs

```diff
@@ -68,75 +83,138 @@ internal class MaterialPropertyPollingFrequency
             {
-                return RenderQueue == PollingFrequency.UpdateContinuously;
-            }
-            else
-            {
-                MaterialPropertyPollingFrequency pollingFrequency;
-                if (PollingFrequencyByMaterialProperty.TryGetValue(new MaterialPropertyKey(materialProperty.ShaderName, materialProperty.propertyName), out pollingFrequency))
+                if (materialProperty.propertyType == MaterialPropertyType.ShaderKeywords)
+                {
+                    return ShaderKeywords == PollingFrequency.UpdateContinuously;
+                }
+                else if (materialProperty.propertyType == MaterialPropertyType.RenderQueue)
+                {
+                    return RenderQueue == PollingFrequency.UpdateContinuously;
+                }
+                else
                 {
-                    switch (pollingFrequency.updateFrequency)
+                    MaterialPropertyPollingFrequency pollingFrequency;
+                    if (PollingFrequencyByMaterialProperty.TryGetValue(new MaterialPropertyKey(materialProperty.ShaderName, materialProperty.propertyName), out pollingFrequency))
                     {
-                        case PollingFrequency.UpdateContinuously:
-                            return true;
-                        case PollingFrequency.UpdateOnceOnStart:
-                            return false;
+                        switch (pollingFrequency.updateFrequency)
+                        {
+                            case PollingFrequency.UpdateContinuously:
+                                return true;
+                            case PollingFrequency.UpdateOnceOnStart:
+                                return false;
+                        }
                     }
-                }
 
-                // If we have a parent, check the parent to see if the parent has an explicit override list
-                if (materialProperties == PollingFrequency.InheritFromParent && parentParameters != null)
-                {
-                    return parentParameters.ShouldUpdateMaterialProperty(materialProperty);
+                    // If we have a parent, check the parent to see if the parent has an explicit override list
+                    if (materialProperties != PollingFrequency.InheritFromParent || parentParameters == null)
+                    {
+                        return materialProperties == PollingFrequency.UpdateContinuously;
+                    }
                 }
+            }
 
-                return materialProperties == PollingFrequency.UpdateContinuously;
+            // Stop the timer before calling parent function
+            return parentParameters.ShouldUpdateMaterialProperty(materialProperty);
+        }
+
+        public bool IsShaderSupported(string shader)
+        {
+            if (ShaderKeywords == PollingFrequency.UpdateContinuously)
+            {
+                return true;
             }

```

Comment:
I don't think this is a correct usage of ShaderKeywords. ShaderKeywords is meant to control how frequently the shaderKeywords of a material are synced (if those keywords could possibly change), not whether the entire shader is supported.